test/InSkillPurchase.spec.js

```diff
@@ -0,0 +1,338 @@
+        expect(reply.msg.directives[0].type).to.equal('Connections.SendRequest');
+        expect(reply.msg.directives[0].name).to.equal('Upsell');
+        expect(reply.msg.directives[0].payload.InSkillProduct.productId).to.equal('2');
+        expect(reply.msg.directives[0].payload.upsellMessage).to.equal(upsellMessage);
+        expect(reply.msg.directives[0].token).to.equal('startState');
+        expect(reply.session.attributes.state).to.equal('die');
+        expect(reply.toJSON().response.shouldEndSession).to.equal(true);
+      });
+  });
+
+  it('should not send ISP directives on invalid endpoint', () => {
+    const event = {
+      request: {
+        type: 'IntentRequest',
+        locale: 'en-US',
+        intent: {
+          name: 'BuyIntent',
+          slots: {
+            productName: { name: 'productName', value: 'shield' },
+          },
+        },
+      },
+      session: {
+        new: true,
+        application: {
+          applicationId: 'appId',
+        },
+      },
+      context: {
+        System: {
+          apiEndpoint: 'https://api.fe.amazonalexa.com',
+          apiAccessToken: 'apiAccessToken',
+        },
+      },
+    };
+
+    const stateMachineSkill = new StateMachineSkill({ variables, views });
+    stateMachineSkill.onIntent('BuyIntent', (alexaEvent) => {
+      if (!alexaEvent.isp.isAllowed()) {
+        return { reply: 'ISP.Invalid', to: 'entry' };
+      }
+
+      return { to: 'entry' };
+    });
+
+    return stateMachineSkill.execute(event)
+      .then((reply) => {
+        expect(reply.msg.statements[0]).to.equal('To do In Skill Purchases, you need to link your Amazon account to the US market.');
+        expect(reply.msg.reprompt).to.equal('Can you try again?');
+        expect(reply.msg.directives).to.be.empty;
+        expect(reply.session.attributes.state).to.equal('entry');
+        expect(reply.toJSON().response.shouldEndSession).to.equal(false);
+      });
+  });
+
+  it('should handle ACCEPTED purchase result', () => {
+    const event = {
+      request: {
+        type: 'Connections.Response',
+        requestId: 'string',
+        timestamp: 'string',
+        name: 'Buy',
+        status: {
+          code: '200',
+          message: 'OK',
+        },
+        payload: {
+          purchaseResult: 'ACCEPTED',
+          productId: 'string',
+          message: 'optional additional message',
+        },
+        token: 'firstState',
+      },
+      session: {
+        new: true,
+        application: {
+          applicationId: 'appId',
+        },
+      },
+    };
+
+    const stateMachineSkill = new StateMachineSkill({ variables, views });
+    stateMachineSkill.onState('firstState', () => ({}));
+    stateMachineSkill.onSessionStarted((alexaEvent) => {
+      alexaEvent.model.flag = 1;
+    });
+
+    stateMachineSkill.onIntent('ConnectionsResponse', (alexaEvent) => {
+      if (alexaEvent.isp.getPayload().purchaseResult === 'ACCEPTED') {
```

Comment:
not sure how much better is this over just `alexaEvent.request.payload.purchaseResult`

Reply:
Yeah, wasn't 100% sure, will remove it